pkg/gds/models/v1/contacts.go

```diff
@@ -0,0 +1,202 @@
+package models
+
+import (
+	"errors"
+	"fmt"
+	"time"
+
+	pb "github.com/trisacrypto/trisa/pkg/trisa/gds/models/v1beta1"
+	"google.golang.org/protobuf/types/known/anypb"
+)
+
+const (
+	TechnicalContact      = "technical"
+	AdministrativeContact = "administrative"
+	LegalContact          = "legal"
+	BillingContact        = "billing"
+)
+
+type contactType struct {
+	contact *pb.Contact
+	kind    string
+}
+
+// Returns True if a Contact is not nil and has an email address.
+func ContactHasEmail(contact *pb.Contact) bool {
+	return contact != nil && contact.Email != ""
+}
+
+// Returns True if a Contact is verified.
+func ContactVerified(contact *pb.Contact) (verified bool, err error) {
+	if _, verified, err = GetContactVerification(contact); err != nil {
+		return false, err
+	}
+	return verified, nil
+}
+
+// Returns a standardized order of iterating through contacts.
+func contactOrder(contacts *pb.Contacts) []*contactType {
+	return []*contactType{
+		{contact: contacts.Technical, kind: TechnicalContact},
+		{contact: contacts.Administrative, kind: AdministrativeContact},
+		{contact: contacts.Legal, kind: LegalContact},
+		{contact: contacts.Billing, kind: BillingContact},
+	}
+}
+
+// Returns a function which iterates over the verified contacts in a Contacts object.
+func IterContacts(contacts *pb.Contacts, requireEmail bool) func() (*pb.Contact, string) {
+	all := contactOrder(contacts)
+	i := 0
+	return func() (contact *pb.Contact, kind string) {
+		// Return the next contact that exists or nil to indicate end of iteration.
+		for i < len(all) {
+			contact = all[i].contact
+			kind = all[i].kind
+			i++
+			if contact != nil && (!requireEmail || ContactHasEmail(contact)) {
+				return contact, kind
+			}
+		}
+		return nil, ""
+	}
```

Comment:
This is a very neat implementation, and I admit - not one I've seen before for iteration in Go. I'll be honest that it threw me and I had to read it a few times to fully grok what was going on. I've been playing with it a little bit and I think that I get everything that's happening. 

I honestly feel like this deserves some more commentary, but honestly I don't know what to say!